### ISSUE TITLE
Provide easy way to set the minimum donation amount

### ISSUE THREAD

**Issue by ericnicolaas** [Charitable/Charitable#298]
Currently, there isn't an easy way to set the minimum donation amount. I think we could remedy this in a couple ways. 

First, we could provide an easy way to do this programatically. It's already possible to invalidate donations under a certain amount (see https://github.com/Charitable/library/blob/master/donation-form/set-minimum-donation-amount.php), but this works server-side; it would be nice if this also worked client-side so that server-side request can be avoided. 

Second, assuming we have the above, it would be nice to be able to set the minimum donation amount globally, for all campaigns, using a setting in the admin. A nice to have would be to also be able to override this on a per-campaign basis.

**Comment by ericnicolaas** [comment]
Tagging this as #add-on as I think the second idea might be better suited to an add-on.

**Comment by ericnicolaas** [comment]
Done. To programatically set the minimum donation amount:

```
function ed_charitable_set_minimum_donation_amount() {
   return 2;
}

add_filter( 'charitable_minimum_donation_amount', 'ed_charitable_set_minimum_donation_amount' );
```

Note that for any amount over 0, the minimum must be met or exceeded. i.e. In the example above, a $2 donation will be considered valid. When the minimum is 0 (the default), the minimum must be exceeded. i.e. A $0 is not valid.

To allow a donation of $0, the `charitable_permit_0_donation` can be used (this has been available since 1.0.0 I'm pretty sure):

```
add_filter( 'charitable_permit_0_donation', '__return_true' );
```
